src/withObs.js

```diff
@@ -1,14 +1,10 @@
+import wrap from './utils/wrap';
 import createHelper from './createHelper';
 import mapObs from './mapObs';
 
-const withObs = input =>
-  mapObs(observables => ({
-    ...observables,
-    ...(
-      typeof input === 'function'
-        ? input(observables)
-        : input
-    ),
-  }));
+const withObs = obsMapper => mapObs((obs, props$) => ({
+  ...obs,
+  ...wrap(obsMapper)(obs, props$),
```

Comment:
Hopefully [this](https://github.com/neoziro/recompact/pull/32#issuecomment-261204896) will convince you.